mfr/extensions/zip/render.py

```diff
@@ -0,0 +1,53 @@
+import os
+import zipfile
+
+import markupsafe
+from mako.lookup import TemplateLookup
+
+from mfr.core import extension
+from mfr.core.utils import sizeof_fmt
+
+
+class ZipRenderer(extension.BaseRenderer):
+
+    TEMPLATE = TemplateLookup(
+        directories=[
+            os.path.join(os.path.dirname(__file__), 'templates')
+        ]).get_template('viewer.mako')
+
+    def render(self):
+        zip_file = zipfile.ZipFile(self.file_path, 'r')
+
+        return self.TEMPLATE.render(zipped_filenames=self.format_zip(zip_file))
+
+    def format_zip(self, zip_file):
+
+        if type(zip_file) != zipfile.ZipFile:
+            return 'This is not a valid zip file.'
+
+        filelist = [file for file in zip_file.filelist if not file.filename.startswith('__MACOSX')]
+
+        if not filelist:
+            return 'This zip file is empty.'
```

Comment:
Please raise errors instead of returning a string.  E.g.: https://github.com/CenterForOpenScience/modular-file-renderer/blob/develop/mfr/extensions/jasp/render.py#L107

Reply:
This is not really the same use case, there's no error it's just an actually empty zip which is a valid though rare possiblity.